benchmark/blas/blas.cpp

```diff
@@ -0,0 +1,540 @@
+        return y_->get_size()[0] * y_->get_size()[1] * 2;
+    }
+
+    gko::size_type get_memory() const override
+    {
+        return y_->get_size()[0] * y_->get_size()[1] * sizeof(etype);
+    }
+
+    void run() override { y_->compute_norm2(lend(alpha_)); }
+
+private:
+    std::unique_ptr<gko::matrix::Dense<etype>> alpha_;
+    std::unique_ptr<gko::matrix::Dense<etype>> y_;
+};
+
+
+class ApplyOperation : public BenchmarkOperation {
+public:
+    ApplyOperation(std::shared_ptr<const gko::Executor> exec, gko::size_type n,
+                   gko::size_type k, gko::size_type m, gko::size_type stride_A,
+                   gko::size_type stride_B, gko::size_type stride_C)
+    {
+        A_ = gko::matrix::Dense<etype>::create(exec, gko::dim<2>{n, k},
+                                               stride_A);
+        B_ = gko::matrix::Dense<etype>::create(exec, gko::dim<2>{k, m},
+                                               stride_B);
+        C_ = gko::matrix::Dense<etype>::create(exec, gko::dim<2>{n, m},
+                                               stride_C);
+        A_->fill(1);
+        B_->fill(1);
+    }
+
+    gko::size_type get_flops() const override
+    {
+        return A_->get_size()[0] * A_->get_size()[1] * B_->get_size()[1] * 2;
+    }
+
+    gko::size_type get_memory() const override
+    {
+        return (A_->get_size()[0] * A_->get_size()[1] +
+                B_->get_size()[0] * B_->get_size()[1] +
+                C_->get_size()[0] * C_->get_size()[1]) *
+               sizeof(etype);
+    }
+
+    void run() override { B_->apply(lend(A_), lend(C_)); }
+
+private:
+    std::unique_ptr<gko::matrix::Dense<etype>> A_;
+    std::unique_ptr<gko::matrix::Dense<etype>> B_;
+    std::unique_ptr<gko::matrix::Dense<etype>> C_;
+};
+
+
+class AdvancedApplyOperation : public BenchmarkOperation {
+public:
+    AdvancedApplyOperation(std::shared_ptr<const gko::Executor> exec,
+                           gko::size_type n, gko::size_type k, gko::size_type m,
+                           gko::size_type stride_A, gko::size_type stride_B,
+                           gko::size_type stride_C)
+    {
+        A_ = gko::matrix::Dense<etype>::create(exec, gko::dim<2>{n, k},
+                                               stride_A);
+        B_ = gko::matrix::Dense<etype>::create(exec, gko::dim<2>{k, m},
+                                               stride_B);
+        C_ = gko::matrix::Dense<etype>::create(exec, gko::dim<2>{n, m},
+                                               stride_C);
+        alpha_ = gko::matrix::Dense<etype>::create(exec, gko::dim<2>{1, 1});
+        beta_ = gko::matrix::Dense<etype>::create(exec, gko::dim<2>{1, 1});
+        A_->fill(1);
+        B_->fill(1);
+        alpha_->fill(1);
+        beta_->fill(1);
+    }
+
+    gko::size_type get_flops() const override
+    {
+        return A_->get_size()[0] * A_->get_size()[1] * B_->get_size()[1] * 2 +
+               C_->get_size()[0] * C_->get_size()[1] * 3;
```

Comment:
this is 2 multiplication + 1 addition from alpha*(AB)_ij + beta * C_ij, right?

Reply:
exactly